tests/virt_autotest/guest_upgrade_run.pm

```diff
@@ -119,21 +119,15 @@ sub run {
     my $self = shift;
     my $timeout = get_var('MAX_TEST_TIME', '36000') + 10;
     my $upload_log_name = 'guest-upgrade-logs';
-    if (is_s390x) {
-        #ues die_on_timeout=> 0 as workaround for s390x test during call script_run, refer to poo#106765
-        script_run("echo \"Debug info: max_test_time is $timeout\"", die_on_timeout => 0);
-    } else {
-        script_run("echo \"Debug info: max_test_time is $timeout\"");
-    }
+    script_run("timeout 20 echo \"Debug info: max_test_time is $timeout\"");
     # Modify source configuration file sources.* of virtauto-data pkg on host
     # to use openqa daily build installer repo and module repo for guests,
     # and it will be copied into guests to be used during guest upgrade test
     repl_module_in_sourcefile();
 
     $self->{'package_name'} = 'Guest Upgrade Test';
     if (is_s390x) {
-        #ues die_on_timeout=> 0 as workaround for s390x test during call script_run, refer to poo#106765
-        script_run "[ -d /var/log/qa/ctcs2 ] && rm -r /var/log/qa/ctcs2 ; [ -d /var/lib/libvirt/images/prj4_guest_upgrade ] && rm -r /var/lib/libvirt/images/prj4_guest_upgrade /tmp/full_guest_upgrade_test-* /tmp/kill_zypper_procs-* /tmp/update-guest-*", die_on_timeout => 0;
+        script_run "[ -d /var/log/qa/ctcs2 ] && rm -r /var/log/qa/ctcs2 ; [ -d /var/lib/libvirt/images/prj4_guest_upgrade ] && rm -r /var/lib/libvirt/images/prj4_guest_upgrade /tmp/full_guest_upgrade_test-* /tmp/kill_zypper_procs-* /tmp/update-guest-*";
```

Comment:
This does not seem equivalent. S390 needs to continue when timeout happens.

Reply:
`lpar_cmd()`? @guoxuguang

Reply:
the referenced ticket poo#106765 is a bit older so I assume that this workaround is maybe not necessary anymore. An alternative is to call

```suggestion
        enter_cmd "[ -d /var/log/qa/ctcs2 ] && rm -r /var/log/qa/ctcs2 ; [ -d /var/lib/libvirt/images/prj4_guest_upgrade ] && rm -r /var/lib/libvirt/images/prj4_guest_upgrade /tmp/full_guest_upgrade_test-* /tmp/kill_zypper_procs-* /tmp/update-guest-*; echo DONE";
        wait_serial 'DONE', 30;
```

Reply:
Please see my updated version.